Manishdas/everything to get started with mongo db/add

manishdasa100 opened the pull request:
## Closes #2 

<Put a x between the square brackets. Make sure you get them all>

### Checklist: 

- [x] Added activities for learners
- [x] Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x] Clean and well documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

amoghrajesh reviewed (changes requested):
Hey @manishdasa100!
The content is good and the activities are quite engaging too. I like the way you have provided hints and curious cats sections here and there to keep the learner in a constant thinking mindset.

Although I have some changes I need to request from you:

1. Since we follow a learn by doing approach, we don't need so much content for installation and setup of MongoDB. Primary reason being that MongoDB is not very hard to install and I think learners can figure it out on their own. So, I request you to cut down that content significantly and provide links for reading and trying out rather.
2. You can shorten the content of 'Bits and Bytes of Mongodb' by writing points that are shorter.
3. I think you can merge the first 4 activities into 1 activity or 2 activities as it is just an intro. Also, have lesser reading and more doing in these activities.
4. In the 7th Activity, instead of providing all the commands and stuff, you can make it more engaging by showing one example and other examples can be questions posed to the user. Also, maybe don't mention $gte, $gle, and all these, these can just be questions and they can learn its usage using these questions.

My final feedback would be - Use points instead of using paragraphs ad learners can tend to get overwhelmed by a lot of content at once.

Great work, man! I hope you make these changes ASAP so we can review it again 😉

amoghrajesh commented on Everything to get started with mongoDB/README.md:
````diff
@@ -0,0 +1,162 @@
+Commands that are executed against the current database are executed against the db object. db is an object reference to the current database object. You can run a bunch of functions associated with the db objects. To know the functions available type ```db.help()```. Upon doing so it will list all the functions that you can run against the db object. Try picking a function from the list and run  ...
+
+### Activity 2 ###
+Notice how you are usign () after a function name while running the commands? Now run the same commands but this time without the () after the function name. What difference do you see in the output? Try doing the same thing with some other functions. Do you see the pattern repeating? If you are familiar with javascript, you will see a resemblance because mongo shell runs on javascript. So next t ...
````
Look out for typos - using*

amoghrajesh commented on Everything to get started with mongoDB/README.md:
```diff
@@ -0,0 +1,162 @@
+    Congratulations if you could come up with the correct answer.
+</details>
+
+## Summery ##
+With this we have come to the end of this microbyte. We haven't looked at a little more advance stuffs like update, upserts, odering, indexing and paging which pushes the learning time beyond one hour for this microbyte. However, we did get MongoDB up and running, looked at insert and remove commands(there isn't much more than what we have seen). We also introduced find and learned how to perform ...
```
Make this into points instead of paragraphs. Would be easier to read

manishdasa100 commented:
Thanks @amoghrajesh.
Updates:
1. Corrected typo.
2. Merged Activities 1,2,3 and 4 into two activities.
3. Summarized in bullet points.
4. Moved the procedure to create a mongo cluster into a google so that whoever wants to can refer. I have kept the steps to run mongoDB locally otherwise I have to additionally mention about mongo shell.

Ak-Shaw commented on Everything to get started with mongoDB/README.md:
```diff
@@ -0,0 +1,137 @@
+    Congratulations if you could come up with the correct answer.
+</details>
+
+## Summery ##
```
LGTM! Just a typo though. :sweat_smile: It's SUMMARY!

Anyways, I like your content. Let's see what the other reviewers have to day.

amoghrajesh reviewed (changes requested):
Hey @manishdasa100 !
I really liked your work but I would like one more change:
You can cut down activity 5 into smaller tasks so that learner doesn't feel like all the data has been given in one place. For eg: Tell to write a query that selects all users whose age in greater than 50 years without giving the usage of $gte. 
This will push the reader to search online and come up with the solution. I hope you are getting what I mean. Make this for some operators like -> $gte, $ne and rest you can just make a mention off.
Apart from this, great content 👍🏽

manishdasa100 commented:
Thanks @Ak-Shaw @amoghrajesh 
Modified Activity 5 and corrected typo.

manishdasa100 commented:
@amoghrajesh I have modified the activity 5 as per you said. Can you verify?

amoghrajesh commented:
Hey @manishdasa100!
We will review the content asap 😄 
Thanks for changing it

amoghrajesh reviewed (approved):
Hey @manishdasa100!
I just reviewed the content and it is much better than the earlier versions right now.
I really like the way you put Activity 5 together.
Thanks for making the timely changes.
While other reviewers go through it, start thinking of some learner issues that can be raised on top of this one. I see a lot of scope for raising learner issues on this particular one.

Good work, man 😄

Kevinpaulose05 reviewed (approved):
Awesome work @manishdasa100 . U have shown passion and patience by making this wonderful microbyte. One typo in the end "To summerise this microbyte". Rest all I well and good. Try coming up with more microbytes . Two things to do now:
1. Try and open issues for learners (if there exists any scope) for them to contribute to ur microbyte
2. Showcase ur microbyte using this [template](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) on slack microbyte-showcase channel

Happy Contributing!!

YourAKShaw commented:
@all-contributors kindly add @manishdasa100 for:

- [x] code
- [x] content
- [x] ideas
- [x] doc

YourAKShaw commented:
@all-contributors kindly add @Kevinpaulose05 for review and eventOrganizing.
